src/main/java/com/synopsys/arc/jenkinsci/plugins/extratoolinstallers/installers/RarExtractionInstaller.java

```diff
@@ -0,0 +1,157 @@
+/*
+ * The MIT License
+ *
+ * Copyright (c) 2009, Sun Microsystems, Inc.
+ * Martin Hjelmqvist
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+package com.synopsys.arc.jenkinsci.plugins.extratoolinstallers.installers;
+
+import hudson.Extension;
+import hudson.FilePath;
+import hudson.ProxyConfiguration;
+import hudson.Functions;
+import hudson.model.Node;
+import hudson.model.TaskListener;
+import hudson.remoting.VirtualChannel;
+import hudson.tools.ToolInstallation;
+import hudson.tools.ToolInstallerDescriptor;
+import hudson.util.FormValidation;
+import jenkins.MasterToSlaveFileCallable;
+import java.io.File;
+import java.io.IOException;
+import java.net.HttpURLConnection;
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.net.URLConnection;
+import org.kohsuke.stapler.DataBoundConstructor;
+import org.kohsuke.stapler.QueryParameter;
+
+/**
+ * Installs a tool by downloading and unpacking a RAR file. 
+ * The installer is inspired by "ZipExtractionInstaller" from the Custom Tools plugin. 
+ * Limitation: Extracts only a single RAR file. A tool archive split into multiple RAR volumes will be not fully extracted.
+ *
+ * @author Martin Hjelmqvist <martin@hjelmqvist.eu>.
+ */
+public class RarExtractionInstaller extends AbstractExtraToolInstaller {
+
+    /**
+     * Messages displayed when configuring the tool in Jenkins. 
+     * TODO: Move this to Messages.java generated by the localizer.
+     */
+    private static final String RAR_EXTRACTION_INSTALLER_DISPLAY_NAME = "Extract *.rar";
```

Comment:
These strings should be kept in messages. See https://wiki.jenkins-ci.org/display/JENKINS/Internationalization